packages/@lwc/engine-core/src/framework/base-lightning-element.ts

```diff
@@ -495,65 +474,84 @@ LightningElement.prototype = {
 
 const queryAndChildGetterDescriptors: PropertyDescriptorMap = create(null);
 
-const childGetters: Array<
-    [
-        keyof HTMLElement,
-        keyof Pick<
-            Renderer,
-            | 'getChildren'
-            | 'getChildNodes'
-            | 'getFirstChild'
-            | 'getFirstElementChild'
-            | 'getLastChild'
-            | 'getLastElementChild'
-        >
-    ]
-> = [
-    ['children', 'getChildren'],
-    ['childNodes', 'getChildNodes'],
-    ['firstChild', 'getFirstChild'],
-    ['firstElementChild', 'getFirstElementChild'],
-    ['lastChild', 'getLastChild'],
-    ['lastElementChild', 'getLastElementChild'],
+// The reason we don't just call `import * as renderer from '../renderer'` here is that the bundle size
+// is smaller if we reference each function individually. Otherwise Rollup will create one big frozen
+// object representing the renderer, with a lot of methods we don't actually need.
+const childGetters = <const>[
+    'children',
+    'childNodes',
+    'firstChild',
+    'firstElementChild',
+    'lastChild',
+    'lastElementChild',
 ];
 
+function getChildGetter(methodName: typeof childGetters[number]) {
+    switch (methodName) {
+        case 'children':
+            return getChildren;
+        case 'childNodes':
+            return getChildNodes;
+        case 'firstChild':
+            return getFirstChild;
+        case 'firstElementChild':
+            return getFirstElementChild;
+        case 'lastChild':
+            return getLastChild;
+        case 'lastElementChild':
+            return getLastElementChild;
+    }
+}
+
 // Generic passthrough for child getters on HTMLElement to the relevant Renderer APIs
-for (const [elementProp, rendererMethod] of childGetters) {
-    queryAndChildGetterDescriptors[elementProp] = {
+for (const childGetter of childGetters) {
+    queryAndChildGetterDescriptors[childGetter] = {
         get(this: LightningElement) {
             const vm = getAssociatedVM(this);
-            const { elm, renderer } = vm;
+            const { elm } = vm;
 
             if (process.env.NODE_ENV !== 'production') {
-                warnIfInvokedDuringConstruction(vm, elementProp);
+                warnIfInvokedDuringConstruction(vm, childGetter);
             }
 
-            return renderer[rendererMethod](elm);
+            return getChildGetter(childGetter)(elm);
```

Comment:
Instead of calling `getChildGetter` every time the getter is invoked we should cache the method in the loop:

```ts
for (const childGetter of childGetters) {
    const method = getChildGetter(childGetter);

    queryAndChildGetterDescriptors[childGetter] = {
        get(this: LightningElement) {
            /*... */
            return method(elm);
        },
        configurable: true,
        enumerable: true,
    };
}
```

Reply:
Unfortunately we can't. Because we have to wait for `set*` to be called, we actually can't cache the method until the first time someone calls the getter (otherwise it's undefined).

We could cache it after the first access, but I'm not sure it would be a perf improvement.

Reply:
No, it doesn't worth adding extra code here.